### PR TITLE
Converting Aspect to JSON-LD

### LINKED ISSUE #644: [Task] Represent Aspect Model in JSON-LD
For custom implementation projects, a json-ld representation of an aspect model (as opposed to rdf ttl) is beneficial in order to read aspect models using json based tools

**Describe the solution you'd like**

**Describe alternatives you've considered**
riot cli from apache jena could be used, but avoiding this dependency is more convenient

**Additional context**

### PR DESCRIPTION
**Pull request by Yauhenikapl**
## Description

Representation Aspect to JSON-LD

Fixes #644

## Type of change

Please delete options that are not relevant.
- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: core/esmf-aspect-model-document-generators/src/test/java/org/eclipse/esmf/aspectmodel/generator/jsonld/AspectModelJsonLdGeneratorTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package org.eclipse.esmf.aspectmodel.generator.jsonld;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.eclipse.esmf.aspectmodel.urn.AspectModelUrn;
+import org.eclipse.esmf.metamodel.Aspect;
+import org.eclipse.esmf.test.TestAspect;
+import org.eclipse.esmf.test.TestResources;
+
+import com.github.jsonldjava.core.JsonLdError;
```
Comment (atextor) [inline core/esmf-aspect-model-document-generators/src/test/java/org/eclipse/esmf/aspectmodel/generator/jsonld/AspectModelJsonLdGeneratorTest.java:10]: Is this intentional?

File: core/esmf-aspect-model-document-generators/src/test/java/org/eclipse/esmf/aspectmodel/generator/jsonld/AspectModelJsonLdGeneratorTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.junit.jupiter.api.Test;
+
+class AspectModelJsonLdGeneratorTest {
+
+   @Test
+   void generateTest() throws JsonLdError {
```
Comment (atextor) [inline core/esmf-aspect-model-document-generators/src/test/java/org/eclipse/esmf/aspectmodel/generator/jsonld/AspectModelJsonLdGeneratorTest.java:16]: Can we remove the `throws`?

File: documentation/developer-guide/modules/tooling-guide/pages/maven-plugin.adoc [hunk 3]
```diff
@@ -523,6 +523,52 @@ Configuration Properties:
 | `outputDirectory` | The path to the directory where the generated JSON payload will be written to. | `String` | none | {ok}
 |===
 
+=== Generating Sample JSON-LD
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/maven-plugin.adoc:526]: This does not create sample data for an Aspect Model, but a JSON-LD representation of the model itself.
```suggestion
=== Generating JSON-LD representation of an Aspect Model
```

File: documentation/developer-guide/modules/tooling-guide/pages/maven-plugin.adoc [hunk 4]
```diff
@@ -523,6 +523,52 @@ Configuration Properties:
...
+
+The `generateJsonLd` goal generates a valid sample JSON-LD for a given Aspect Model as it
+could be returned by an Aspect that implements that Aspect Model. The default life cycle phase for
+the goal is `generate-resources`.
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/maven-plugin.adoc:530]:
```suggestion
The `generateJsonLd` goal generates a JSON-LD represention of a given Aspect Model. The default life cycle phase for the goal is `generate-resources`.
```

File: documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc [hunk 5]
```diff
@@ -146,6 +146,9 @@ The available options and their meaning can also be seen in the help text of the
 .3+| [[aspect-to-json]] aspect <model> to json | Generate example JSON payload data for an Aspect Model                      | `samm aspect AspectModel.ttl to json`
                                    | _--output, -o_ : output file path (default: stdout)                                     |
                                    | _--custom-resolver_ : use an external resolver for the resolution of the model elements |
+.3+| [[aspect-to-jsonld]] aspect <model> to jsonld | Generate example JSON-LD data for an Aspect Model                      | `samm aspect AspectModel.ttl to jsonld`
+                                   | _--output, -o_ : output file path (default: stdout)                                     |
+                                   | _--custom-resolver_ : use an external resolver for the resolution of the model elements |
```
Comment (atextor) [inline documentation/developer-guide/modules/tooling-guide/pages/samm-cli.adoc:151]:
```suggestion
.3+| [[aspect-to-jsonld]] aspect <model> to jsonld | Generate JSON-LD representation of an Aspect Model                      | `samm aspect AspectModel.ttl to jsonld`
                                   | _--output, -o_ : output file path (default: stdout)                                     |
                                   | _--custom-resolver_ : use an external resolver for the resolution of the model elements |
```

File: tools/esmf-aspect-model-maven-plugin/src/main/java/org/eclipse/esmf/aspectmodel/GenerateJsonLd.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+      final Set<Aspect> aspects = loadAspects();
+      try {
+         for ( final Aspect context : aspects ) {
+            final AspectModelToJsonLdGenerator generator = new AspectModelToJsonLdGenerator( context );
+            generator.generateJsonLd( name -> getOutputStreamForFile( name + ".jsonld", outputDirectory ) );
+         }
+      } catch ( final IOException exception ) {
+         throw new MojoExecutionException( "Could not generate JSON-LD.", exception );
+      }
+      LOG.info( "Successfully generated example JSON-LD for Aspect Models." );
```
Comment (atextor) [inline tools/esmf-aspect-model-maven-plugin/src/main/java/org/eclipse/esmf/aspectmodel/GenerateJsonLd.java:46]:
```suggestion
      LOG.info( "Successfully generated JSON-LD for Aspect Models." );
```

File: tools/samm-cli/src/main/java/org/eclipse/esmf/aspect/to/AspectToJsonLdCommand.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+import org.eclipse.esmf.LoggingMixin;
+import org.eclipse.esmf.ResolverConfigurationMixin;
+import org.eclipse.esmf.aspect.AspectToCommand;
+import org.eclipse.esmf.aspectmodel.generator.jsonld.AspectModelToJsonLdGenerator;
+import org.eclipse.esmf.exception.CommandException;
+
+import picocli.CommandLine;
+
+@CommandLine.Command( name = AspectToJsonLdCommand.COMMAND_NAME,
+      description = "Generate JSON-LD specification for an Aspect Model",
```
Comment (atextor) [inline tools/samm-cli/src/main/java/org/eclipse/esmf/aspect/to/AspectToJsonLdCommand.java:28]:
```suggestion
      description = "Generate JSON-LD representation of an Aspect Model",
```

**Review by atextor — CHANGES_REQUESTED** [review]

File: core/esmf-aspect-model-document-generators/src/test/java/org/eclipse/esmf/aspectmodel/generator/jsonld/AspectModelJsonLdGeneratorTest.java (hunk 2 above)
Comment (Yauhenikapl) [reply core/esmf-aspect-model-document-generators/src/test/java/org/eclipse/esmf/aspectmodel/generator/jsonld/AspectModelJsonLdGeneratorTest.java:16]: Yes, 100%

**Review by atextor — APPROVED** [review]

**Pull request merged**